weldx/core/generic_series.py

```diff
@@ -259,6 +259,10 @@ def __eq__(self, other):
             return False
         return self.data_array.identical(other._obj)
 
+    def __hash__(self):
```

Comment:
are there any performance implications for this change? This will trigger equality comparisons for set and dict operations.

I've seen deepsource complaining about this. But can't we redirect the hash function to XArray (if they offer one)?

Reply:
Actually, mutable objects shouldn't be hashable at all. But there is a feature request for XArray to implement this any ways. https://github.com/pydata/xarray/issues/4738

Reply:
the way I see it this should have been the implicit behavior anyway (because we implement `__eq__`) but somehow deepsource didn't pick it up

Reply:
I think if the equality operator is defined, and the hash method is just being derived, the object will still be hashable. So this is not just a cosmetic change to satisfy a linter. 
Prior 0.6.2 this raised:
``` python
import weldx
x = weldx.core.MathematicalExpression("3*x")
gs = weldx.core.GenericSeries(x)
hash(gs)
```
```TypeError: __hash__ method should return an integer```